test/Microsoft.OData.ModelBuilder.Tests/Vocabularies/Core/RevisionsConfigurationTests.cs

```diff
@@ -0,0 +1,198 @@
+
+            var revisionsBuilder = modelBuilder
+                .EntitySet<Customer>("Customers")
+                .HasDeprecation(Description, Version, Date, RemovalDate);
+
+            var model = modelBuilder.GetServiceModel();
+            var container = model.SchemaElements.OfType<IEdmEntityContainer>().SingleOrDefault();
+            var customers = container.EntitySets().SingleOrDefault(e => e.Name == "Customers");
+
+            var term = model.FindTerm("Org.OData.Core.V1.Revisions");
+            var annotation = customers.VocabularyAnnotations(model).FirstOrDefault(a => a.Term == term);
+            Assert.NotNull(annotation);
+
+            var annotationValue = annotation.Value as EdmCollectionExpression;
+            Assert.NotNull(annotationValue);
+
+            var revisionsProperties = Assert.Single(annotationValue.Elements) as EdmRecordExpression;
+            Assert.NotNull(revisionsProperties);
+
+            var descriptionNameValue = GetRecordValue<EdmStringConstant>(revisionsProperties, "Description");
+            Assert.NotNull(descriptionNameValue);
+
+            Assert.Equal(Description, descriptionNameValue.Value);
+
+            var revisionKindValue = GetRecordValue<EdmEnumMemberExpression>(revisionsProperties, "Kind");
+            Assert.NotNull(revisionKindValue);
+
+            Assert.Equal(Kind.ToString(), revisionKindValue.EnumMembers.FirstOrDefault().Name);
+
+        }
+
+        [Fact]
+        public void RevisionsCanBeAddedToEntityType()
+        {
+            var modelBuilder = new ODataModelBuilder();
+            var entityTypeConfiguration = modelBuilder.Add_Customer_EntityType();
+
+            var revisionsBuilder = modelBuilder
+               .EntityType<Customer>()
+               .HasRevisions(Kind, Description, Version);
+
+            var model = modelBuilder.GetServiceModel();
+            var entityType = model.SchemaElements.OfType<IEdmEntityType>().Single();
+
+            var term = model.FindTerm("Org.OData.Core.V1.Revisions");
+            var annotation = entityType.VocabularyAnnotations(model).FirstOrDefault(a => a.Term == term);
+            Assert.NotNull(annotation);
+
+            var annotationValue = annotation.Value as EdmCollectionExpression;
+            Assert.NotNull(annotationValue);
+
+            var revisionsProperties = Assert.Single(annotationValue.Elements) as EdmRecordExpression;
+            Assert.NotNull(revisionsProperties);
+
+            var descriptionValue = GetRecordValue<EdmStringConstant>(revisionsProperties, "Description");
+            Assert.NotNull(descriptionValue);
+
+            Assert.Equal(Description, descriptionValue.Value);
+
+            var revisionKindValue = GetRecordValue<EdmEnumMemberExpression>(revisionsProperties, "Kind");
+            Assert.NotNull(revisionKindValue);
+
+            Assert.Equal(Kind.ToString(), revisionKindValue.EnumMembers.FirstOrDefault().Name);
```

Comment:
It seems most of these code are duplicated in each test cases. Could we make a verify method to wrapper them?